src/Features/Core/Portable/Extensions/IExtensionAssemblyLoaderProvider.cs

```diff
@@ -54,7 +54,7 @@ private sealed class DefaultExtensionAssemblyLoaderProvider(HostWorkspaceService
             // These lines should always succeed.  If they don't, they indicate a bug in our code that we want
             // to bubble out as it must be fixed.
             var analyzerAssemblyLoaderProvider = _workspaceServices.GetRequiredService<IAnalyzerAssemblyLoaderProvider>();
-            var analyzerAssemblyLoader = analyzerAssemblyLoaderProvider.CreateNewShadowCopyLoader();
+            var analyzerAssemblyLoader = analyzerAssemblyLoaderProvider.CreateNewShadowCopyLoader(collectOnDispose: true);
```

Comment:
was there another place where we need to set this to true, e.g. for analyzers and generators in the OOP process?  cc @CyrusNajmabadi

Reply:
Looking.